guppylang/checker/linearity_checker.py

```diff
@@ -317,13 +317,13 @@ def _reassign_inout_args(self, func_ty: FunctionType, call: AnyCall) -> None:
             if InputFlags.Inout in inp.flags:
                 match arg:
                     case PlaceNode(place=place):
-                        self._reassign_single_inout_arg(place, arg)
+                        self._reassign_single_inout_arg(place, place.defined_at or arg)
```

Comment:
drive-by: Improve leak errors by keeping track of the original place where a variable was defined if possible